### PR TITLE
SX: disallow nested pseudo classes/elements

### PR DESCRIPTION
**Pull request by mrtnzlml**
This change disallows cases like:

```text
{
  ':hover': {
    'color': 'red',
    ':active': {
      color: 'blue'
    }
  }
}
```

Instead, you can write this:

```text
{
  ':hover': {
    color: 'red',
  },
  ':active': {
    color: 'blue'
  }
}
```

### CODE REVIEW

File: src/sx/src/StyleCollector.js [hunk 1]
```diff
@@ -61,6 +61,9 @@ class StyleCollector {
           }
         } else if (maybeName.startsWith(':')) {
           // pseudo type (:hover, ::after, ...)
+          for (const key of Object.keys(maybeValue)) {
+            invariant(key.startsWith(':') === false, 'Nested pseudo classes are not allowed.');
```
Comment (tbergq) [inline src/sx/src/StyleCollector.js:65]: what about`::after:hover`?

Comment (mrtnzlml) [reply src/sx/src/StyleCollector.js:65]: Do you mean when you write it as one selector? I actually didn't try it but I don't see a problem with that. It just cannot be nested. Will add some tests for it as well.

Comment (tbergq) [reply src/sx/src/StyleCollector.js:65]: Aha, of course, it is not nested in this case 🤦

**Review by tbergq — APPROVED** [review]

File: src/sx/src/StyleCollector.js (hunk 1 above)
Comment (mrtnzlml) [reply src/sx/src/StyleCollector.js:65]: I added a test to cover this scenario:

```text
exports[`matches expected output: pseudo-classes-nested-2.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    ":hover::after": {
      "content": "this is OK"
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
._2BDpfz:hover::after {
  content: "this is OK";
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="_2BDpfz"

`;
```

**Review by tbergq — APPROVED** [review]

**Pull request merged**